### ISSUE TITLE
Issue with SDK file path after High Sierra update

### ISSUE THREAD

**Issue by CrashBurnRepeat** [barche/QML.jl#50]
I tried to build QML after updating MacOS to High Sierra and I got a build error that seems to be related to the following warning:
```
INFO: Changing Directory to /Users/crashburnrepeat/.julia/v0.6/QML/deps/builds/qmlwrap
CMake Warning at /Users/crashburnrepeat/.julia/v0.6/Homebrew/deps/usr/Cellar/cmake/3.10.0/share/cmake/Modules/Platform/Darwin-Initialize.cmake:121 (message):
  Ignoring CMAKE_OSX_SYSROOT value:

   /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.12.sdk

  because the directory does not exist.
```
Manually changing the value in the `CMakeCache.txt` file to `MacOSX10.13.sdk`, this warning disappears and the package builds properly. Since `MacOSX10.13.sdk` seems to be a symbolic link to `MacOSX.sdk`, QML also builds correctly when that is the set path. Is there a reason why the OS version is kept cache like this, and why it isn't updated when QML detects that it's no longer a valid path? Might it be easier just to reference the version-less sdk location?

**Comment by barche** [comment]
Yep, same CMake problem as in CxxWrap, unfortunately I don't think there's an easy way to automate deleting the cache. Maybe in Pkg3 it will be easier to distinguish between "user" and "developer" installations, and always force a full rebuild for users when `Pkg.build` is ran (for developers this would be too time consuming).

**Comment by CrashBurnRepeat** [comment]
I'm still relatively new to the Julia community... do you feel this would be something worth raising an issue on JuliaLang?

**Comment by barche** [comment]
Well, this issue only comes up on packages that compile their own C/C++ code, which is not that common, so I don't think it's worth raising a general issue. Also, it is my understanding that Pkg3 will offer a way to distinguish between the usage and development environment, so maybe we will be able to use that as part of the solution. Another option is to force a rebuild by default (i.e. nuke the build and usr directories always upon build) and provide an option for developers to disable this.

**Comment by ufechner7** [comment]
Perhaps this can be closed now that 0.8 is released?
